src/client/datascience/types.ts

```diff
@@ -153,33 +141,34 @@ export interface INotebook {
 }
 
 // Options for connecting to a notebook provider
-export type ConnectNotebookProviderOptions = {
+export interface ConnectNotebookProviderOptions {
     getOnly?: boolean;
     disableUI?: boolean;
     localOnly?: boolean;
     token?: CancellationToken;
     resource: Resource;
-};
+}
 
 export interface INotebookServerOptions {
+    /**
+     * Undefined when connecting to local Jupyter (in case Raw kernels aren't supported)
+     */
     uri?: string;
```

Comment:
This was important for me, hence added comments.
Was looking at perf and came here.